From b45ea1289d6b3a693391bc087f35d144c1c04f32 Mon Sep 17 00:00:00 2001
From: Sami Shalayel <sami.shalayel@qt.io>
Date: Wed, 17 Sep 2025 14:38:44 +0200
Subject: [PATCH 059/239] CRA: review src/qmlls

Mark all files as significant.

Use reason:trusted-sources on files that spawn QProcess or parse data:
* qdochtmlparser.cpp parses content obtained from the QHelpEngine,
  which is a trusted source
* qqmlcodemodel is used to spawn QProcesses that runs CMake (which is
  a trusted source) with arguments from
  QQmlLSUtils::cmakeBuildCommand(const QString &path) (which is also a
  trusted source) or trusted hard-coded arguments.

Pick-to: 6.9 6.8
Fixes: QTBUG-136201
Change-Id: I3f2ff1dfc208224baff6fe88b337fe24c44b599a
Reviewed-by: Ulf Hermann <ulf.hermann@qt.io>
(cherry picked from commit 1cd950dbbb0a0df0b1b8a7bf401326795f1e6952)
---
 src/qmlls/qdochtmlparser.cpp                | 1 +
 src/qmlls/qdochtmlparser_p.h                | 1 +
 src/qmlls/qlanguageserver.cpp               | 1 +
 src/qmlls/qlanguageserver_p.h               | 1 +
 src/qmlls/qlanguageserver_p_p.h             | 1 +
 src/qmlls/qlspcustomtypes_p.h               | 2 ++
 src/qmlls/qmllsmain.cpp                     | 1 +
 src/qmlls/qmllsmain_p.h                     | 1 +
 src/qmlls/qqmlbasemodule_p.h                | 1 +
 src/qmlls/qqmlcodemodel.cpp                 | 1 +
 src/qmlls/qqmlcodemodel_p.h                 | 1 +
 src/qmlls/qqmlcompletioncontextstrings.cpp  | 1 +
 src/qmlls/qqmlcompletionsupport.cpp         | 1 +
 src/qmlls/qqmlcompletionsupport_p.h         | 1 +
 src/qmlls/qqmlfindusagessupport.cpp         | 1 +
 src/qmlls/qqmlfindusagessupport_p.h         | 1 +
 src/qmlls/qqmlformatting.cpp                | 1 +
 src/qmlls/qqmlformatting_p.h                | 1 +
 src/qmlls/qqmlgotodefinitionsupport.cpp     | 1 +
 src/qmlls/qqmlgotodefinitionsupport_p.h     | 1 +
 src/qmlls/qqmlgototypedefinitionsupport.cpp | 1 +
 src/qmlls/qqmlgototypedefinitionsupport_p.h | 1 +
 src/qmlls/qqmlhighlightsupport.cpp          | 1 +
 src/qmlls/qqmlhighlightsupport_p.h          | 1 +
 src/qmlls/qqmlhover.cpp                     | 1 +
 src/qmlls/qqmlhover_p.h                     | 1 +
 src/qmlls/qqmllanguageserver.cpp            | 1 +
 src/qmlls/qqmllanguageserver_p.h            | 1 +
 src/qmlls/qqmllintsuggestions.cpp           | 1 +
 src/qmlls/qqmllintsuggestions_p.h           | 1 +
 src/qmlls/qqmllscompletion.cpp              | 1 +
 src/qmlls/qqmllscompletion_p.h              | 1 +
 src/qmlls/qqmllscompletionplugin.cpp        | 1 +
 src/qmlls/qqmllscompletionplugin_p.h        | 1 +
 src/qmlls/qqmllshelpplugininterface.cpp     | 1 +
 src/qmlls/qqmllshelpplugininterface_p.h     | 1 +
 src/qmlls/qqmllshelputils.cpp               | 1 +
 src/qmlls/qqmllshelputils_p.h               | 1 +
 src/qmlls/qqmllsplugin_p.h                  | 1 +
 src/qmlls/qqmllsutils.cpp                   | 1 +
 src/qmlls/qqmllsutils_p.h                   | 1 +
 src/qmlls/qqmlrangeformatting.cpp           | 1 +
 src/qmlls/qqmlrangeformatting_p.h           | 1 +
 src/qmlls/qqmlrenamesymbolsupport.cpp       | 1 +
 src/qmlls/qqmlrenamesymbolsupport_p.h       | 1 +
 src/qmlls/qqmlsemantictokens.cpp            | 1 +
 src/qmlls/qqmlsemantictokens_p.h            | 1 +
 src/qmlls/qtextblock.cpp                    | 1 +
 src/qmlls/qtextblock_p.h                    | 1 +
 src/qmlls/qtextcursor.cpp                   | 1 +
 src/qmlls/qtextcursor_p.h                   | 1 +
 src/qmlls/qtextdocument.cpp                 | 1 +
 src/qmlls/qtextdocument_p.h                 | 1 +
 src/qmlls/qtextsynchronization.cpp          | 1 +
 src/qmlls/qtextsynchronization_p.h          | 1 +
 src/qmlls/qworkspace.cpp                    | 1 +
 src/qmlls/qworkspace_p.h                    | 1 +
 57 files changed, 58 insertions(+)

diff --git a/src/qmlls/qdochtmlparser.cpp b/src/qmlls/qdochtmlparser.cpp
index 6450e7b0af..8380c296eb 100644
--- a/src/qmlls/qdochtmlparser.cpp
+++ b/src/qmlls/qdochtmlparser.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:trusted-sources
 
 #include <qdochtmlparser_p.h>
 #include <QtCore/qregularexpression.h>
diff --git a/src/qmlls/qdochtmlparser_p.h b/src/qmlls/qdochtmlparser_p.h
index d09f2f882e..14ad24abf6 100644
--- a/src/qmlls/qdochtmlparser_p.h
+++ b/src/qmlls/qdochtmlparser_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QDOCHTMLEXTRACTOR_P_H
 #define QDOCHTMLEXTRACTOR_P_H
diff --git a/src/qmlls/qlanguageserver.cpp b/src/qmlls/qlanguageserver.cpp
index d90d650a60..23d5dc62a1 100644
--- a/src/qmlls/qlanguageserver.cpp
+++ b/src/qmlls/qlanguageserver.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qlanguageserver_p_p.h"
 
diff --git a/src/qmlls/qlanguageserver_p.h b/src/qmlls/qlanguageserver_p.h
index f59ee1ecd1..af94afb10e 100644
--- a/src/qmlls/qlanguageserver_p.h
+++ b/src/qmlls/qlanguageserver_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QLANGUAGESERVER_P_H
 #define QLANGUAGESERVER_P_H
diff --git a/src/qmlls/qlanguageserver_p_p.h b/src/qmlls/qlanguageserver_p_p.h
index 792a1503bf..8b956e63e1 100644
--- a/src/qmlls/qlanguageserver_p_p.h
+++ b/src/qmlls/qlanguageserver_p_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QLANGUAGESERVER_P_P_H
 #define QLANGUAGESERVER_P_P_H
diff --git a/src/qmlls/qlspcustomtypes_p.h b/src/qmlls/qlspcustomtypes_p.h
index e299388b79..113b83f017 100644
--- a/src/qmlls/qlspcustomtypes_p.h
+++ b/src/qmlls/qlspcustomtypes_p.h
@@ -1,5 +1,7 @@
 // Copyright (C) 2022 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
+
 #ifndef QLSPCUSTOMTYPES_P_H
 #define QLSPCUSTOMTYPES_P_H
 
diff --git a/src/qmlls/qmllsmain.cpp b/src/qmlls/qmllsmain.cpp
index e41c7da57a..5c11206673 100644
--- a/src/qmlls/qmllsmain.cpp
+++ b/src/qmlls/qmllsmain.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2025 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qmllsmain_p.h"
 
diff --git a/src/qmlls/qmllsmain_p.h b/src/qmlls/qmllsmain_p.h
index 9ea9b2382b..ada1616110 100644
--- a/src/qmlls/qmllsmain_p.h
+++ b/src/qmlls/qmllsmain_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2025 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QMLLSMAIN_P_H
 #define QMLLSMAIN_P_H
diff --git a/src/qmlls/qqmlbasemodule_p.h b/src/qmlls/qqmlbasemodule_p.h
index 294b38fa40..e1e798f1f7 100644
--- a/src/qmlls/qqmlbasemodule_p.h
+++ b/src/qmlls/qqmlbasemodule_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLBASEMODULE_P_H
 #define QQMLBASEMODULE_P_H
diff --git a/src/qmlls/qqmlcodemodel.cpp b/src/qmlls/qqmlcodemodel.cpp
index 1ccfcb9a14..a375052809 100644
--- a/src/qmlls/qqmlcodemodel.cpp
+++ b/src/qmlls/qqmlcodemodel.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:trusted-sources
 
 #include "qqmlcodemodel_p.h"
 #include "qqmllsplugin_p.h"
diff --git a/src/qmlls/qqmlcodemodel_p.h b/src/qmlls/qqmlcodemodel_p.h
index ba2a2781ae..34ce08862d 100644
--- a/src/qmlls/qqmlcodemodel_p.h
+++ b/src/qmlls/qqmlcodemodel_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLCODEMODEL_P_H
 #define QQMLCODEMODEL_P_H
diff --git a/src/qmlls/qqmlcompletioncontextstrings.cpp b/src/qmlls/qqmlcompletioncontextstrings.cpp
index 5fc2006661..329e0d0ea8 100644
--- a/src/qmlls/qqmlcompletioncontextstrings.cpp
+++ b/src/qmlls/qqmlcompletioncontextstrings.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmlcompletioncontextstrings_p.h"
 
diff --git a/src/qmlls/qqmlcompletionsupport.cpp b/src/qmlls/qqmlcompletionsupport.cpp
index 632a5e902a..b7da33d9ef 100644
--- a/src/qmlls/qqmlcompletionsupport.cpp
+++ b/src/qmlls/qqmlcompletionsupport.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmlcompletionsupport_p.h"
 #include "qqmllsutils_p.h"
diff --git a/src/qmlls/qqmlcompletionsupport_p.h b/src/qmlls/qqmlcompletionsupport_p.h
index 3d7fa03d94..05a88cbe91 100644
--- a/src/qmlls/qqmlcompletionsupport_p.h
+++ b/src/qmlls/qqmlcompletionsupport_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2018 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLCOMPLETIONSUPPORT_P_H
 #define QQMLCOMPLETIONSUPPORT_P_H
diff --git a/src/qmlls/qqmlfindusagessupport.cpp b/src/qmlls/qqmlfindusagessupport.cpp
index 4ec712b22d..f390cab427 100644
--- a/src/qmlls/qqmlfindusagessupport.cpp
+++ b/src/qmlls/qqmlfindusagessupport.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmlfindusagessupport_p.h"
 #include "qqmllsutils_p.h"
diff --git a/src/qmlls/qqmlfindusagessupport_p.h b/src/qmlls/qqmlfindusagessupport_p.h
index ee4803ea93..006c04d56b 100644
--- a/src/qmlls/qqmlfindusagessupport_p.h
+++ b/src/qmlls/qqmlfindusagessupport_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QMLFINDUSAGESUPPORT_P_H
 #define QMLFINDUSAGESUPPORT_P_H
diff --git a/src/qmlls/qqmlformatting.cpp b/src/qmlls/qqmlformatting.cpp
index 19fe3ddabb..01a8fc02c7 100644
--- a/src/qmlls/qqmlformatting.cpp
+++ b/src/qmlls/qqmlformatting.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include <qqmlformatting_p.h>
 #include <qqmlcodemodel_p.h>
diff --git a/src/qmlls/qqmlformatting_p.h b/src/qmlls/qqmlformatting_p.h
index 8bd423e988..edba96bce1 100644
--- a/src/qmlls/qqmlformatting_p.h
+++ b/src/qmlls/qqmlformatting_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLFORMATTING_P_H
 #define QQMLFORMATTING_P_H
diff --git a/src/qmlls/qqmlgotodefinitionsupport.cpp b/src/qmlls/qqmlgotodefinitionsupport.cpp
index e27f3decf2..fb94eee98b 100644
--- a/src/qmlls/qqmlgotodefinitionsupport.cpp
+++ b/src/qmlls/qqmlgotodefinitionsupport.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmlgotodefinitionsupport_p.h"
 #include "qqmllsutils_p.h"
diff --git a/src/qmlls/qqmlgotodefinitionsupport_p.h b/src/qmlls/qqmlgotodefinitionsupport_p.h
index d4553e3b67..61c2177f52 100644
--- a/src/qmlls/qqmlgotodefinitionsupport_p.h
+++ b/src/qmlls/qqmlgotodefinitionsupport_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLGOTODEFINITIONSUPPORT_P_H
 #define QQMLGOTODEFINITIONSUPPORT_P_H
diff --git a/src/qmlls/qqmlgototypedefinitionsupport.cpp b/src/qmlls/qqmlgototypedefinitionsupport.cpp
index b6129ac524..b0e2f0ca23 100644
--- a/src/qmlls/qqmlgototypedefinitionsupport.cpp
+++ b/src/qmlls/qqmlgototypedefinitionsupport.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmlgototypedefinitionsupport_p.h"
 #include "qqmllsutils_p.h"
diff --git a/src/qmlls/qqmlgototypedefinitionsupport_p.h b/src/qmlls/qqmlgototypedefinitionsupport_p.h
index a8af07fd91..ff4f13befa 100644
--- a/src/qmlls/qqmlgototypedefinitionsupport_p.h
+++ b/src/qmlls/qqmlgototypedefinitionsupport_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QMLGOTOTYPEDEFINITIONSUPPORT_P_H
 #define QMLGOTOTYPEDEFINITIONSUPPORT_P_H
diff --git a/src/qmlls/qqmlhighlightsupport.cpp b/src/qmlls/qqmlhighlightsupport.cpp
index a74eaa3d02..8b7c85ad62 100644
--- a/src/qmlls/qqmlhighlightsupport.cpp
+++ b/src/qmlls/qqmlhighlightsupport.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include <qqmlhighlightsupport_p.h>
 
diff --git a/src/qmlls/qqmlhighlightsupport_p.h b/src/qmlls/qqmlhighlightsupport_p.h
index ffb29e9df9..80cc036496 100644
--- a/src/qmlls/qqmlhighlightsupport_p.h
+++ b/src/qmlls/qqmlhighlightsupport_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLHIGHLIGHTSUPPORT_P_H
 #define QQMLHIGHLIGHTSUPPORT_P_H
diff --git a/src/qmlls/qqmlhover.cpp b/src/qmlls/qqmlhover.cpp
index f7f578d2be..a8612feb44 100644
--- a/src/qmlls/qqmlhover.cpp
+++ b/src/qmlls/qqmlhover.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmlhover_p.h"
 #include <QtQmlLS/private/qqmllshelputils_p.h>
diff --git a/src/qmlls/qqmlhover_p.h b/src/qmlls/qqmlhover_p.h
index ba964adc8b..8db2b36d7a 100644
--- a/src/qmlls/qqmlhover_p.h
+++ b/src/qmlls/qqmlhover_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLHOVER_P_H
 #define QQMLHOVER_P_H
diff --git a/src/qmlls/qqmllanguageserver.cpp b/src/qmlls/qqmllanguageserver.cpp
index de0051d923..71222d750e 100644
--- a/src/qmlls/qqmllanguageserver.cpp
+++ b/src/qmlls/qqmllanguageserver.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmllanguageserver_p.h"
 #include "qtextsynchronization_p.h"
diff --git a/src/qmlls/qqmllanguageserver_p.h b/src/qmlls/qqmllanguageserver_p.h
index 11f59bee16..bc034b8e0f 100644
--- a/src/qmlls/qqmllanguageserver_p.h
+++ b/src/qmlls/qqmllanguageserver_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLLANGUAGESERVER_P_H
 #define QQMLLANGUAGESERVER_P_H
diff --git a/src/qmlls/qqmllintsuggestions.cpp b/src/qmlls/qqmllintsuggestions.cpp
index c18910e87d..e15616c3cc 100644
--- a/src/qmlls/qqmllintsuggestions.cpp
+++ b/src/qmlls/qqmllintsuggestions.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmllintsuggestions_p.h"
 
diff --git a/src/qmlls/qqmllintsuggestions_p.h b/src/qmlls/qqmllintsuggestions_p.h
index 437cf1064f..11bec1729b 100644
--- a/src/qmlls/qqmllintsuggestions_p.h
+++ b/src/qmlls/qqmllintsuggestions_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QMLLINTSUGGESTIONS_P_H
 #define QMLLINTSUGGESTIONS_P_H
diff --git a/src/qmlls/qqmllscompletion.cpp b/src/qmlls/qqmllscompletion.cpp
index 444a9a83b8..dd83237baf 100644
--- a/src/qmlls/qqmllscompletion.cpp
+++ b/src/qmlls/qqmllscompletion.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmllscompletion_p.h"
 
diff --git a/src/qmlls/qqmllscompletion_p.h b/src/qmlls/qqmllscompletion_p.h
index d9143801c6..529722ef6e 100644
--- a/src/qmlls/qqmllscompletion_p.h
+++ b/src/qmlls/qqmllscompletion_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLLSCOMPLETION_H
 #define QQMLLSCOMPLETION_H
diff --git a/src/qmlls/qqmllscompletionplugin.cpp b/src/qmlls/qqmllscompletionplugin.cpp
index fd47c691f7..0a3d7f347a 100644
--- a/src/qmlls/qqmllscompletionplugin.cpp
+++ b/src/qmlls/qqmllscompletionplugin.cpp
@@ -1,4 +1,5 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmllscompletionplugin_p.h"
diff --git a/src/qmlls/qqmllscompletionplugin_p.h b/src/qmlls/qqmllscompletionplugin_p.h
index 0dde7bec76..8bda88134b 100644
--- a/src/qmlls/qqmllscompletionplugin_p.h
+++ b/src/qmlls/qqmllscompletionplugin_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLLSCOMPLETIONPLUGIN_H
 #define QQMLLSCOMPLETIONPLUGIN_H
diff --git a/src/qmlls/qqmllshelpplugininterface.cpp b/src/qmlls/qqmllshelpplugininterface.cpp
index 62944d69e0..27f09c2ce2 100644
--- a/src/qmlls/qqmllshelpplugininterface.cpp
+++ b/src/qmlls/qqmllshelpplugininterface.cpp
@@ -1,4 +1,5 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmllshelpplugininterface_p.h"
diff --git a/src/qmlls/qqmllshelpplugininterface_p.h b/src/qmlls/qqmllshelpplugininterface_p.h
index 280d7c3e5d..02fdd940db 100644
--- a/src/qmlls/qqmllshelpplugininterface_p.h
+++ b/src/qmlls/qqmllshelpplugininterface_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLLSHELPPLUGININTERFACE_H
 #define QQMLLSHELPPLUGININTERFACE_H
diff --git a/src/qmlls/qqmllshelputils.cpp b/src/qmlls/qqmllshelputils.cpp
index 54469e9b71..f52d4a8bf1 100644
--- a/src/qmlls/qqmllshelputils.cpp
+++ b/src/qmlls/qqmllshelputils.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmllshelputils_p.h"
 
diff --git a/src/qmlls/qqmllshelputils_p.h b/src/qmlls/qqmllshelputils_p.h
index 8d8814522d..8991ff9daa 100644
--- a/src/qmlls/qqmllshelputils_p.h
+++ b/src/qmlls/qqmllshelputils_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLLSHELPUTILS_P_H
 #define QQMLLSHELPUTILS_P_H
diff --git a/src/qmlls/qqmllsplugin_p.h b/src/qmlls/qqmllsplugin_p.h
index 07699ce2c5..b6d4c28c0e 100644
--- a/src/qmlls/qqmllsplugin_p.h
+++ b/src/qmlls/qqmllsplugin_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QMLLSPLUGIN_P_H
 #define QMLLSPLUGIN_P_H
diff --git a/src/qmlls/qqmllsutils.cpp b/src/qmlls/qqmllsutils.cpp
index 58b62de749..997c6e155a 100644
--- a/src/qmlls/qqmllsutils.cpp
+++ b/src/qmlls/qqmllsutils.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmllsutils_p.h"
 
diff --git a/src/qmlls/qqmllsutils_p.h b/src/qmlls/qqmllsutils_p.h
index 6d91fd8866..2de40bb879 100644
--- a/src/qmlls/qqmllsutils_p.h
+++ b/src/qmlls/qqmllsutils_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QLANGUAGESERVERUTILS_P_H
 #define QLANGUAGESERVERUTILS_P_H
diff --git a/src/qmlls/qqmlrangeformatting.cpp b/src/qmlls/qqmlrangeformatting.cpp
index 807e5db577..5459e855fb 100644
--- a/src/qmlls/qqmlrangeformatting.cpp
+++ b/src/qmlls/qqmlrangeformatting.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include <qqmlrangeformatting_p.h>
 #include <qqmlcodemodel_p.h>
diff --git a/src/qmlls/qqmlrangeformatting_p.h b/src/qmlls/qqmlrangeformatting_p.h
index 5c51ea4f12..4b42804e85 100644
--- a/src/qmlls/qqmlrangeformatting_p.h
+++ b/src/qmlls/qqmlrangeformatting_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLRANGEFORMATTING_P_H
 #define QQMLRANGEFORMATTING_P_H
diff --git a/src/qmlls/qqmlrenamesymbolsupport.cpp b/src/qmlls/qqmlrenamesymbolsupport.cpp
index 2a094dd791..a0aa0bc6f7 100644
--- a/src/qmlls/qqmlrenamesymbolsupport.cpp
+++ b/src/qmlls/qqmlrenamesymbolsupport.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qqmllsutils_p.h"
 #include "qqmlrenamesymbolsupport_p.h"
diff --git a/src/qmlls/qqmlrenamesymbolsupport_p.h b/src/qmlls/qqmlrenamesymbolsupport_p.h
index 0f1d8be252..c2448f0c15 100644
--- a/src/qmlls/qqmlrenamesymbolsupport_p.h
+++ b/src/qmlls/qqmlrenamesymbolsupport_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLRENAMESYMBOLSUPPORT_P_H
 #define QQMLRENAMESYMBOLSUPPORT_P_H
diff --git a/src/qmlls/qqmlsemantictokens.cpp b/src/qmlls/qqmlsemantictokens.cpp
index 734b178d79..c3ff530354 100644
--- a/src/qmlls/qqmlsemantictokens.cpp
+++ b/src/qmlls/qqmlsemantictokens.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include <qqmlsemantictokens_p.h>
 
diff --git a/src/qmlls/qqmlsemantictokens_p.h b/src/qmlls/qqmlsemantictokens_p.h
index c73fead176..cefc04c478 100644
--- a/src/qmlls/qqmlsemantictokens_p.h
+++ b/src/qmlls/qqmlsemantictokens_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QQMLSEMANTICTOKENS_P_H
 #define QQMLSEMANTICTOKENS_P_H
diff --git a/src/qmlls/qtextblock.cpp b/src/qmlls/qtextblock.cpp
index beca4a7880..bd0d9c370d 100644
--- a/src/qmlls/qtextblock.cpp
+++ b/src/qmlls/qtextblock.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qtextblock_p.h"
 #include "qtextdocument_p.h"
diff --git a/src/qmlls/qtextblock_p.h b/src/qmlls/qtextblock_p.h
index f7bb039460..247ae2a664 100644
--- a/src/qmlls/qtextblock_p.h
+++ b/src/qmlls/qtextblock_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QTEXTBLOCK_P_H
 #define QTEXTBLOCK_P_H
diff --git a/src/qmlls/qtextcursor.cpp b/src/qmlls/qtextcursor.cpp
index 295ba3144b..d6b0920f0d 100644
--- a/src/qmlls/qtextcursor.cpp
+++ b/src/qmlls/qtextcursor.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qtextcursor_p.h"
 #include "qtextdocument_p.h"
diff --git a/src/qmlls/qtextcursor_p.h b/src/qmlls/qtextcursor_p.h
index 9c9b781d3c..f34645acb8 100644
--- a/src/qmlls/qtextcursor_p.h
+++ b/src/qmlls/qtextcursor_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef TEXTCURSOR_H
 #define TEXTCURSOR_H
diff --git a/src/qmlls/qtextdocument.cpp b/src/qmlls/qtextdocument.cpp
index f69f1b385c..bfe397cb91 100644
--- a/src/qmlls/qtextdocument.cpp
+++ b/src/qmlls/qtextdocument.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qtextdocument_p.h"
 #include "qtextblock_p.h"
diff --git a/src/qmlls/qtextdocument_p.h b/src/qmlls/qtextdocument_p.h
index 5d70f0f651..1d94bb1198 100644
--- a/src/qmlls/qtextdocument_p.h
+++ b/src/qmlls/qtextdocument_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QTEXTDOCUMENT_P_H
 #define QTEXTDOCUMENT_P_H
diff --git a/src/qmlls/qtextsynchronization.cpp b/src/qmlls/qtextsynchronization.cpp
index 0762072939..18c916cdba 100644
--- a/src/qmlls/qtextsynchronization.cpp
+++ b/src/qmlls/qtextsynchronization.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qtextsynchronization_p.h"
 #include "qqmllsutils_p.h"
diff --git a/src/qmlls/qtextsynchronization_p.h b/src/qmlls/qtextsynchronization_p.h
index 2c26b84aea..23a027441a 100644
--- a/src/qmlls/qtextsynchronization_p.h
+++ b/src/qmlls/qtextsynchronization_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QTEXTSYNCHRONIZATION_P_H
 #define QTEXTSYNCHRONIZATION_P_H
diff --git a/src/qmlls/qworkspace.cpp b/src/qmlls/qworkspace.cpp
index 2db04a376c..62e0c74f93 100644
--- a/src/qmlls/qworkspace.cpp
+++ b/src/qmlls/qworkspace.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qworkspace_p.h"
 #include "qqmllanguageserver_p.h"
diff --git a/src/qmlls/qworkspace_p.h b/src/qmlls/qworkspace_p.h
index 3a4520ab8b..71cfd022ad 100644
--- a/src/qmlls/qworkspace_p.h
+++ b/src/qmlls/qworkspace_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QWORKSPACE_P_H
 #define QWORKSPACE_P_H
-- 
2.51.2

